doc/user/content/sql/system-catalog/mz_internal.md

```diff
@@ -1045,80 +1045,82 @@ messages and additional metadata helpful for debugging.
+envelopes must maintain an index. These gauges reset to 0 when the source is
+restarted, as the index must be rehydrated.
 
----
-**Snapshot progress**
+`rehydration_latency` represents the amount of time it took for the source to
+rehydrate its index after the latest restart. It is reset to `NULL` when a
+source is restarted and is populated with a duration after hydration finishes. 
 
-When a source is first created, it must process and initial snapshot of data. `snapshot_records_known` is the full size of that snapshot, and `snapshot_records_staged`
-is how much of that snapshot the source has read so far.
+When a source is first created, it must process an initial snapshot of data.
+`snapshot_records_known` is the total number of records in the snapshot, and
+`snapshot_records_staged` is how many of the records the source has read so far.
 
-The _size_ of the snapshot has a source-defined unit:
+The meaning of record depends on the source:
 - For Kafka sources, its the total number of offsets in the snapshot.
 - For Postgres and MySQL sources, its the number of rows in the snapshot.
 
-Note that when tables are added to Postgres or MySQL sources, this statistics will reset as we snapshot those new tables.
-
-### Gauges
-Gauges never decrease/reset.
-
----
-**Snapshot Completion**
-`snapshot_committed` becomes true when we have fully _committed_ the snapshot for the given source.
-
----
-**Steady-state progress**
+Note that when tables are added to Postgres or MySQL sources,
+`snapshot_records_known` and `snapshot_records_staged` will reset as the source
+snapshots those new tables. The metrics will also reset if the source is
+restarted while the snapshot is in progres.
 
-`offset_known` and `offset_committed` are used to represent the _progress_ a source is making,
-in comparison to its upstream source. They are designed to be turned into _rates_, and compared.
+`snapshot_committed` becomes true when we have fully committed the snapshot for
+the given source. <!-- TODO: when does this reset? -->
 
-`offset_known` is the maximum offset of upstream data knows about. `offset_committed` is the offset that Materialize has committed data up to.
+`offset_known` and `offset_committed` are used to represent the progress a
+source is making relative to its upstream source. `offset_known` is the maximum
+offset in the upstream system that Materialize knows about. `offset_committed`
+is the offset that Materialize has durably ingested. These metrics will never
+decrease over the lifetime of a source.
 
-These statistics have a source-defined unit:
-- For Kafka sources, its the number of offsets.
-- For MySQL sources, its the number of transactions.
-- For Postgres sources, its the number of bytes in its replication stream.
+The meaning of offset depends on the source:
+- For Kafka sources, an offset is the Kafka message offset.
+- For MySQL sources, an offset is the number of transactions committed across all servers in the cluster.
```

Comment:
Q: Should we append with  `(GTID)` --  (https://materializeinc.slack.com/archives/CQM08AP9V/p1733296713840719?thread_ts=1733258115.549209&cid=CQM08AP9V ?

Reply:
It's not actually a GTID. It's weirder than that. A GTID is a combo of a server UUID and a transaction number. Multiple servers can be committing transactions at the same time. So we sum up the transaction numbers across all servers. In the general case, it doesn't really tie back to anything in MySQL land unfortunately. (Although if you're only using one server it'll be the transaction number of that one server.)

Reply:
👍 Thanks for the clarification!